src/memoryprofiler.js

```diff
@@ -35,6 +35,16 @@ var emscriptenMemoryProfiler = {
   // corresponds to the data added to the VFS with --preload-file.
   sizeOfPreRunAllocatedPtr: {},
 
+  resizeMemorySources: [],
```

Comment:
I'm not sure what you mean by "bad" in the PR description?

Adding some comments here to explain why we track these things might help too.

Reply:
By bad I referred to "multiple places in their code that all sbrk() independently to allocate free memory blocks". Because of the lack of virtual address space, e.g. two malloc implementations A and B that both alternate to acquire memory via sbrk() will cause the wasm heap to be split into badly fragmented sections, like `A | B | A | B | B | A | B`, where neither allocator may then be able to satisfy large allocations, as they have limited each others ability to obtain contiguous memory. Even if other allocator freed all its memory and "shut down" (whatever that might mean for the allocator), the division boundaries are permanent as sbrk() does not have any ability to express freeing up the memory.

With a large address space and virtual memory this would not be much of a problem, but without that it can be quite a big source of tough-to-debug fragmentation.